web/static.go

```diff
@@ -73,7 +76,25 @@ func root(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Cache-Control", "no-cache, max-age=31556926, public")
 
 	staticDir, _ := fileutils.FindDir(model.ClientDir)
-	http.ServeFile(w, r, filepath.Join(staticDir, "root.html"))
+	contents, err := os.ReadFile(filepath.Join(staticDir, "root.html"))
+	if err != nil {
+		// TODO: may be we should return an error page here.
+		return
+	}
+
+	newHTML := string(contents)
+	siteName := c.App.Srv().Config().TeamSettings.SiteName
+	if model.TeamSettingsDefaultSiteName != *siteName {
+		mmTitle := fmt.Sprintf("<title>%s</title>", model.TeamSettingsDefaultSiteName)
+		siteNameTitle := fmt.Sprintf("<title>%s</title>", *siteName)
+		reHTML := regexp.MustCompile(mmTitle)
+		if results := reHTML.FindAllString(newHTML, -1); len(results) > 0 {
+			newHTML = reHTML.ReplaceAllLiteralString(newHTML, siteNameTitle)
+		}
+	}
+
+	w.Header().Set("Content-Type", "text/html")
+	fmt.Fprintln(w, newHTML)
```

Comment:
> It seems odd to add this here without modifying or deleting an existing response writing. Where is this response usually written with the default HTML document?

What do you mean exactly @mickmister?

> fmt.Fprintln(w, newHTML)

This is really inefficient, you could use `w.Write([]byte(newHTML))` directly (and maybe check the error?). Doing the modifications on the `[]byte` representation would make it even better.

Reply:
Used []byte for response as @noxer said.

Reply:
> What do you mean exactly @mickmister?

@noxer I mean "Where is the existing call to `w.Write` or an equivalent that is occurring today for this request?"

We are adding these lines here to respond to the request, but there is no removal of an existing response write in this PR. We are just "adding" this here, and I'm wondering if we are "doubling up" on response writes because of this.

Reply:
> This is really inefficient, you could use w.Write([]byte(newHTML)) directly (and maybe check the error?). Doing the modifications on the []byte representation would make it even better.

@noxer I was just thinking about this. Shouldn't we be caching this calculated value so we're not re-computing this for every request?